src/app/hero-list/hero-list.component.ts

```diff
@@ -10,24 +11,13 @@ import { HeroService } from '../hero.service';
   styleUrls: ['./hero-list.component.scss']
 })
 
-export class HeroListComponent implements OnInit {
+export class HeroListComponent implements OnInit{
   heroes: Hero[] = [];
   nextPage: string
   prevPage: string;
   loading: boolean = false;
-
-  getHeroes(pageNumber: number) {
-    this.loading = true;
-    return this.heroService.getHeroes(pageNumber).pipe(
-      tap((res) => {
-      this.heroes = res.results;
-      this.prevPage = res.previous;
-      this.nextPage = res.next;
-    }), finalize(() => {
-      this.loading = false;
-    })
-    );
-  }
+  emptySearch: boolean = false;
```

Comment:
try to not make additional variables for info we already store, in this case this.heroes.length